File: Assignment3/domain/Author.py
from datetime import datetime

class Author:
    def __init__(self):
        self.id: int = None
        self.graph_id: int = None
        self.type: str = None
        self.attributeKey: str = None
        self.attributeValue: str = None
        self.description: str = None
        self. creationDate: datetime = None
        self.username: str = None
        self.isActive: bool = None
    
    def getId(self) -> int:
        return self.id
        
    def setId(self, id: int) -> None:
        self.id = id
        
    def getGraph_id(self) -> int:
        return self.graph_id
        
    def setGraph_id(self, graph_id: int) -> None:
        self.graph_id = graph_id
        
    def getType(self) -> str:
        return self.type
        
    def setType(self, type: str) -> None:
        self.type = type
        
    def getAttributeKey(self) -> str:
        return self.attributeKey
        
    def setAttributeKey(self, attribute_key: str) -> None:
        self.attributeKey = attribute_key
        
    def getAttributeValue(self) -> str:
        return self.attributeValue
        
    def setAttributeValue(self, attribute_value: str) -> None:
        self.attributeValue = attribute_value
        
    def getDescription(self) -> str:
        return self.description
        
    def setDescription(self, description: str) -> None:
        self.description = description
        
    def getCreationDate(self) -> datetime:
        return self.creationDate
        
    def setCreationDate(self, creation_date: datetime) -> None:
        self.creationDate = creation_date
        
    def getUsername(self) -> str:
        return self.username
        
    def setUsername(self, username: str) -> None:
        self.username = username
        
    def getIsActive(self) -> bool:
        return self.isActive
        
    def setIsActive(self, is_active: bool) -> None:
        self.isActive = is_active

File: Assignment3/domainJava/Author.java
package domain;

import java.sql.Timestamp;

/**
 * Represents an Author entity in the system.
 * This class stores information about an author including identification,
 * associated graph, type, attributes, description, creation date, username
 * and active status.
 */
public class Author {

    /** The unique identifier for the author */
    private Integer id;
    /** The associated graph identifier */
    private Integer graph_id;
    /** The type of the author */
    private String type;
    /** The key of the author's attribute */
    private String attributeKey;
    /** The value of the author's attribute */
    private String attributeValue;
    /** The description of the author */
    private String description;
    /** The date when the author was created */
    private Timestamp creationDate;
    /** The username of the author */
    private String username;
    /** Flag indicating if the author is active */
    private Boolean isActive;
    
    /**
     * Gets the unique identifier of the author.
     * 
     * @return the author's ID
     */
    public Integer getId() {
        return id;
    }
    
    /**
     * Sets the unique identifier of the author.
     * 
     * @param id the ID to set
     */
    public void setId(Integer id) {
        this.id = id;
    }
    
    /**
     * Gets the graph identifier associated with the author.
     * 
     * @return the graph ID
     */
    public Integer getGraph_id() {
        return graph_id;
    }
    
    /**
     * Sets the graph identifier associated with the author.
     * 
     * @param graph_id the graph ID to set
     */
    public void setGraph_id(Integer graph_id) {
        this.graph_id = graph_id;
    }
    
    /**
     * Gets the type of the author.
     * 
     * @return the author type
     */
    public String getType() {
        return type;
    }
    
    /**
     * Sets the type of the author.
     * 
     * @param type the author type to set
     */
    public void setType(String type) {
        this.type = type;
    }
    
    /**
     * Gets the attribute key of the author.
     * 
     * @return the attribute key
     */
    public String getAttributeKey() {
        return attributeKey;
    }
    
    /**
     * Sets the attribute key of the author.
     * 
     * @param attributeKey the attribute key to set
     */
    public void setAttributeKey(String attributeKey) {
        this.attributeKey = attributeKey;
    }
    
    /**
     * Gets the attribute value of the author.
     * 
     * @return the attribute value
     */
    public String getAttributeValue() {
        return attributeValue;
    }
    
    /**
     * Sets the attribute value of the author.
     * 
     * @param attributeValue the attribute value to set
     */
    public void setAttributeValue(String attributeValue) {
        this.attributeValue = attributeValue;
    }
    
    /**
     * Gets the description of the author.
     * 
     * @return the description
     */
    public String getDescription() {
        return description;
    }
    
    /**
     * Sets the description of the author.
     * 
     * @param description the description to set
     */
    public void setDescription(String description) {
        this.description = description;
    }
    
    /**
     * Gets the creation date of the author.
     * 
     * @return the creation date
     */
    public Timestamp getCreationDate() {
        return creationDate;
    }
    
    /**
     * Sets the creation date of the author.
     * 
     * @param creationDate the creation date to set
     */
    public void setCreationDate(Timestamp creationDate) {
        this.creationDate = creationDate;
    }
    
    /**
     * Gets the username of the author.
     * 
     * @return the username
     */
    public String getUsername() {
        return username;
    }
    
    /**
     * Sets the username of the author.
     * 
     * @param username the username to set
     */
    public void setUsername(String username) {
        this.username = username;
    }
    
    /**
     * Checks if the author is active.
     * 
     * @return true if the author is active, false otherwise
     */
    public Boolean getIsActive() {
        return isActive;
    }
    
    /**
     * Sets the active status of the author.
     * 
     * @param isActive the active status to set
     */
    public void setIsActive(Boolean isActive) {
        this.isActive = isActive;
    }
}
